R/project-tool.R

```diff
@@ -1522,41 +1523,38 @@ newProject <- function(destPath = NULL)
+            else if (nrow(csvTab) > 0)
             {
-                csvTab <- tryCatch(fread(csvFile, select = c("path", "analysis", "group", "blank"),
-                                         colClasses = "character"),
-                                   error = function(e) FALSE, warning = function(w) FALSE)
-                if (is.logical(csvTab))
-                    rstudioapi::showDialog("Error", "Failed to open/parse selected csv file!", "")
-                else if (nrow(csvTab) > 0)
+              msExts <- MSFileExtensions()
+              msFiles <- normalizePath(listMSFiles(csvTab$path, MSFileFormats()), winslash = "/")
+              msFilesNoExt <- tools::file_path_sans_ext(msFiles)
+              formats <- mapply(csvTab$analysis, csvTab$path, FUN = function(ana, path)
+              {
+                fps <- msFiles[msFilesNoExt == file.path(path, ana)]
+                if (length(fps) == 0)
+                  return("")
+                ret <- sapply(tolower(tools::file_ext(fps)), function(ext)
                 {
-                    msExts <- MSFileExtensions()
-                    msFiles <- normalizePath(listMSFiles(csvTab$path, MSFileFormats()), winslash = "/")
-                    msFilesNoExt <- tools::file_path_sans_ext(msFiles)
-                    formats <- mapply(csvTab$analysis, csvTab$path, FUN = function(ana, path)
-                    {
-                        fps <- msFiles[msFilesNoExt == file.path(path, ana)]
-                        if (length(fps) == 0)
-                            return("")
-                        ret <- sapply(tolower(tools::file_ext(fps)), function(ext)
-                        {
-                            paste0(names(msExts)[sapply(msExts, function(e) ext %in% tolower(e))], collapse = "/")
-                        })
-                        return(paste0(ret, collapse = ", "))
-                    })
-
-                    csvTab[, format := formats]
-                    csvTab <- csvTab[nzchar(format)] # prune unknown files (might have been removed?)
-                    
-                    if (is.null(csvTab[["norm_conc"]])) # older files lack this column
-                        csvTab[, norm_conc := NA_real_]
-                    csvTab$exclude <- FALSE
-                    rvName <- getCurAnaRVName()
-                    rValues[[rvName]] <- rbind(rValues[[rvName]], csvTab)
-                }
+                  paste0(names(msExts)[sapply(msExts, function(e) ext %in% tolower(e))], collapse = "/")
+                })
+                return(paste0(ret, collapse = ", "))
+              })
+              
+              csvTab[, format := formats]
+              csvTab <- csvTab[nzchar(format)] # prune unknown files (might have been removed?)
+              
+              rvName <- getCurAnaRVName()
+              rValues[[rvName]] <- rbind(rValues[[rvName]], csvTab, fill=TRUE)
             }
+          }
```

Comment:
I think all these changes are primarily from a change in indenting? Can you double check this? I use 4 spaces (no tabs) everywhere.